app/models/resource.rb

```diff
@@ -105,6 +105,12 @@ def search account: nil, kind: nil, owner: nil, offset: nil, limit: nil, search:
       scope = scope.textsearch(search) if search
 
       if offset || limit
+        if limit && !numeric?(limit)
+          raise ApplicationController::UnprocessableEntity, "'limit' contains an invalid value. Limit must be an integer greater than 0."
+        end
+        if offset && !numeric?(offset)
+          raise ApplicationController::UnprocessableEntity, "'offset' contains an invalid value. Offset must be an integer."
```

Comment:
why isn't this also "must be an integer greater than 0"? Is offset -3 valid and should we accept that?